pkg/worker/lifecycle.go

```diff
@@ -128,7 +127,14 @@ func (s *Worker) clearContainer(containerId string, request *types.ContainerRequ
 	go func() {
 		// Allow for some time to pass before clearing the container. This way we can handle some last
 		// minute logs or events or if the user wants to inspect the container before it's cleared.
-		time.Sleep(delay)
+		select {
+		case <-time.After(time.Duration(s.config.Worker.TerminationGracePeriod) * time.Second):
+		case <-s.ctx.Done():
+		}
+
+		if err := s.stopContainer(containerId, false); err != nil {
```

Comment:
I'm confused, in the normal case (i.e. no sigterm), wouldn't the container already be stopped here? Why do we need to call stopcontainer

Reply:
That's right. I just updated this to check if the container is still running.